sdk/core/core-http/test/data/TestClient/src/models/mappers.ts

```diff
@@ -5,6 +5,82 @@
  */
 const internalMappers: any = {};
 
+internalMappers.SimpleProduct = {
+  type: {
+    name: "Composite",
+    className: "SimpleProduct",
+    modelProperties: {
+      id: {
+        serializedName: "id",
+        constraints: {},
+        required: true,
+        type: {
+          name: "Number"
+        }
+      },
+      name: {
+        serializedName: "name",
+        required: true,
+        type: {
+          name: "String"
+        }
+      },
+      maxProductDisplayName: {
+        serializedName: "details.max_product_display_name",
+        type: {
+          name: "String"
+        }
+      },
+      capacity: {
+        defaultValue: "Large",
+        isConstant: true,
+        serializedName: "details.max_product_capacity",
+        type: {
+          name: "String"
+        }
+      }
+    }
+  }
+};
+
+internalMappers.SimpleProductConstFirst = {
+  type: {
+    name: "Composite",
+    className: "SimpleProduct",
+    modelProperties: {
+      id: {
+        serializedName: "id",
+        constraints: {},
+        required: true,
+        type: {
+          name: "Number"
+        }
+      },
+      name: {
+        serializedName: "name",
+        required: true,
+        type: {
+          name: "String"
+        }
+      },
+      capacity: {
+        defaultValue: "Large",
+        isConstant: true,
```

Comment:
Does `capacity` need to be the first field in `modelProperties` to be the 1st property read, or does this adequately test the change since it's the first parameter in the flattened `details`?

Reply:
Yes the bug reproduces when it is the first parameter in the flattened details. It doesn't have to be the first parameter overall

Reply:
Added an example in de description to be more clear